zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/immutable/UserState.java

```diff
@@ -16,5 +17,13 @@ public interface UserState {
 
   UserRecord getUser(final String username);
 
+  /**
+   * Returns a user by its key. If no user was found, an empty optional is returned.
+   *
+   * @param userKey the key of the user
+   * @return An optional containing the user if it was found, otherwise an empty optional
+   */
+  Optional<UserRecord> getUser(final long userKey);
```

Comment:
👍 I like this better than returning `null`.
🔧 I'd propose to make this class consistent though regarding fetching users and the response you receive, either the object directly or an `Optional`.

Reply:
To be more precise, we fetch by username and potentially return a `null` instead of an empty `Optional`.

Reply:
Good call 👍